### PR TITLE
feat(mobile): quiz entry screen bb-80

### PR DESCRIPTION
**Pull request by ToyLess78**
Summary:

This PR introduces the Welcome screen feature, addressing issue 80. The welcome screen has been created using the pre-existing styles and components within the project to maintain consistency across the application.

Details:

Implemented the Welcome screen that serves as the entry point for new users.
Utilized existing styles and components to ensure a cohesive look and feel throughout the app.
Integrated navigation to transition smoothly from the welcome screen to the subsequent screens.
Screenshots:

Знімок екрана 2024-08-25 о 14 27 27

Testing:

Verified the UI consistency with the rest of the application.

### CODE REVIEW

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 1]
```diff
@@ -1,12 +1,60 @@
 import React from "react";
 
-import { Text, View } from "~/libs/components/components";
+import { BackgroundWrapper } from "~/libs/components/background-wrapper/background-wrapper";
+import {
```
Comment (anogire) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:4]: BackgroundWrapper should also be imported from ~/libs/components/components

File: apps/mobile/android/app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -4,6 +4,18 @@
...
+        <item name="android:statusBarColor">@android:color/transparent</item>
+        <item name="android:windowLightStatusBar">false</item> <!-- Icons will be dark (black) -->
+
+        <!-- Make navigation bar transparent and icons black -->
+        <item name="android:navigationBarColor">@android:color/transparent</item>
+        <item name="android:windowLightNavigationBar">false</item> <!-- Icons will be dark (black) -->
+
+        <!-- Make window background transparent -->
+        <item name="android:windowBackground">@android:color/transparent</item>
     </style>
```
Comment (anogire) [inline apps/mobile/android/app/src/main/res/values/styles.xml:19]: I think we can remove unnecessary comments.

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 3]
```diff
@@ -1,12 +1,60 @@
...
+					<Image
+						source={
+							require("~/assets/images/balance-wheel.png") as ImageSourcePropType
+						}
+						style={styles.image}
+					/>
+					<Text
+						preset="subheading"
+						style={[styles.text, globalStyles.pb12]}
+						weight="bold"
```
Comment (anogire) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:44]: Let's use globalStyles first, then the rest.

File: apps/mobile/src/screens/welcome-screen/styles.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { StyleSheet } from "react-native";
+
+import { BaseColor } from "~/libs/enums/app/base-color.enum";
```
Comment (fshabanov) [inline apps/mobile/src/screens/welcome-screen/styles.ts:3]:
```suggestion
import { BaseColor } from "~/libs/enums/enums";
```

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 5]
```diff
@@ -1,12 +1,60 @@
...
+					>
+						Answer a few questions to find out which areas of your life are
+						outstanding and which areas you are missing out on
+					</Text>
+					<Button
+						label="Continue"
+						onPress={(): void => {
+							// Handle Continue
+						}}
```
Comment (fshabanov) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:53]: Avoid inline functions

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by revenets — APPROVED** [review]

**Review by anogire — APPROVED** [review]

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 6]
```diff
@@ -1,12 +1,59 @@
...
+	View,
+} from "~/libs/components/components";
+import { globalStyles } from "~/libs/styles/styles";
+import { ImageSourcePropType } from "~/libs/types/types";
+
+import { styles } from "./styles";
+
+const handleContinuePress = (): void => {
+	// Handle Continue
+};
```
Comment (fshabanov) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:19]: Should be inside of component

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx (hunk 6 above)
Comment (ToyLess78) [reply apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:19]: `[js]   17:39  error  Move arrow function 'handleContinuePress' to the outer scope  unicorn/consistent-function-scoping`

File: apps/mobile/src/screens/welcome-screen/welcome-screen.tsx [hunk 7]
```diff
@@ -1,12 +1,60 @@
...
+	Text,
+	View,
+} from "~/libs/components/components";
+import { useCallback } from "~/libs/hooks/hooks";
+import { globalStyles } from "~/libs/styles/styles";
+import { type ImageSourcePropType } from "~/libs/types/types";
+
+import { styles } from "./styles";
 
 const WelcomeScreen: React.FC = () => {
```
Comment (revenets) [inline apps/mobile/src/screens/welcome-screen/welcome-screen.tsx:17]: As the design has changed please, rename it to `QuizEntryScreen`.

**Review by revenets — CHANGES_REQUESTED** [review]

**Pull request closed without merging**